app/components/templates/form.tsx

```diff
@@ -0,0 +1,285 @@
+          "File size is too big"
+        )
+        .refine(
+          (val: File) => val.type === "application/pdf",
+          "Only .pdf is accepted"
+        ),
+    })
+    .merge(base),
+  z
+    .object({
+      isEdit: z.literal("true"),
+      pdf: z
+        .any()
+        .refine(
+          (val: File) =>
+            val.type !== "application/octet-stream" ||
+            val.size <= MAX_FILE_SIZE,
+          "File size is too big"
+        )
+        .refine(
+          (val: File) =>
+            val.type === "application/octet-stream" ||
+            val.type === "application/pdf",
+          "Only .pdf is accepted"
+        ),
+    })
+    .merge(base),
+]);
+
+interface Props {
+  name?: Template["name"];
+  description?: Template["description"];
+  type?: Template["type"];
+  signatureRequired?: Template["signatureRequired"];
+  pdfUrl?: Template["pdfUrl"];
+  pdfSize?: Template["pdfSize"];
+  pdfName?: Template["pdfName"];
+  isEdit?: boolean;
+}
+
+export const TemplateForm = ({
+  name = "",
+  description = "",
+  type = TemplateType.BOOKINGS,
+  signatureRequired = false,
+  pdfSize,
+  pdfUrl,
+  pdfName,
+  isEdit = false,
+}: Props) => {
+  const navigation = useNavigation();
+  const zo = useZorm("NewQuestionWizardScreen", NewTemplateFormSchema);
+  const disabled = isFormProcessing(navigation.state);
+  const [, validateFile] = useAtom(validateFileAtom);
+
+  const [, updateTitle] = useAtom(updateDynamicTitleAtom);
+  const [selectedType, setSelectedType] = useState<TemplateType>(
+    type || TemplateType.BOOKINGS
+  );
+  const [pdf, setPdf] = useState<File | null>(null);
+
+  const handleFileChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const files = e.target.files;
+      if (!files) return;
+
+      const file = files[0];
+
+      // We don't want to update the state if the file is
+      // more than 5 MB
+      if (file.size > MAX_FILE_SIZE) return;
+
+      setPdf(files[0]);
+      validateFile(e);
+    },
+    [validateFile]
+  );
+
+  return (
+    <Form
+      ref={zo.ref}
+      method="post"
+      className="flex w-full flex-col gap-2"
+      encType="multipart/form-data"
+    >
+      <FormRow rowLabel="Name" className="border-b-0 pb-0" required={true}>
+        <Input
+          label="Name"
+          hideLabel
+          name={zo.fields.name()}
+          disabled={disabled}
+          error={zo.errors.name()?.message}
+          autoFocus
+          onChange={updateTitle}
+          className="w-full"
+          defaultValue={name || ""}
+          placeholder="Booking Arrangement 2023"
+          required={true}
+        />
+      </FormRow>
+      <FormRow rowLabel="Type" className="border-b-0 pb-0" required={true}>
+        <Select
+          name={zo.fields.type()}
+          defaultValue={selectedType}
+          disabled={disabled}
+          onValueChange={(val) =>
+            setSelectedType(
+              val.toString() === TemplateType.BOOKINGS
```

Comment:
Maybe we should drop bookings for now so we can release custody first without being blocked by the bookings signature.

Reply:
uh, so what should i change in this one?

Reply:
We have bookings now so this comment is invalid :)